docs/src/modules/components/ApiPage.js

```diff
@@ -1,439 +0,0 @@
-              </span>
-            </td>
-            <td
-              align="left"
-              dangerouslySetInnerHTML={{
-                __html:
-                  classDescriptions[className] &&
-                  classDescriptions[className].description
-                    .replace(/{{conditions}}/, classDescriptions[className].conditions)
-                    .replace(/{{nodeName}}/, classDescriptions[className].nodeName),
-              }}
-            />
-          </tr>
-        ))}
-      </tbody>
-    </table>
-  );
-}
-
-ClassesTable.propTypes = {
-  classDescriptions: PropTypes.object.isRequired,
-  componentStyles: PropTypes.object.isRequired,
-};
-
-function getTranslatedHeader(t, header) {
-  const translations = {
-    demos: t('api-docs.demos'),
-    import: t('api-docs.import'),
-    'component-name': t('api-docs.componentName'),
-    props: t('api-docs.props'),
-    slots: t('api-docs.slots'),
-    inheritance: t('api-docs.inheritance'),
-    css: 'CSS',
-  };
-
-  // TODO Drop runtime type-checking once we type-check this file
-  if (!translations.hasOwnProperty(header)) {
-    throw new TypeError(
-      `Unable to translate header '${header}'. Did you mean one of '${Object.keys(
-        translations,
-      ).join("', '")}'`,
-    );
-  }
-
-  return translations[header] || header;
-}
-
-function Heading(props) {
-  const { hash, level: Level = 'h2' } = props;
-  const t = useTranslate();
-
-  return (
-    <Level id={hash}>
-      {getTranslatedHeader(t, hash)}
-      <a aria-labelledby={hash} className="anchor-link" href={`#${hash}`} tabIndex={-1}>
-        <svg>
-          <use xlinkHref="#anchor-link-icon" />
-        </svg>
-      </a>
-    </Level>
-  );
-}
-
-Heading.propTypes = {
-  hash: PropTypes.string.isRequired,
-  level: PropTypes.string,
-};
-
-export default function ApiPage(props) {
-  const { descriptions, disableAd = false, pageContent } = props;
-  const t = useTranslate();
-  const userLanguage = useUserLanguage();
-
-  const {
-    cssComponent,
-    demos,
-    filename,
-    forwardsRefTo,
-    inheritance,
-    name: componentName,
-    props: componentProps,
-    spread,
-    slots,
-    styles: componentStyles,
-    packages,
-  } = pageContent;
-
-  const defaultPropsLink = '/material-ui/customization/theme-components/#theme-default-props';
-  const styleOverridesLink = '/material-ui/customization/theme-components/#theme-style-overrides';
-
-  const {
-    componentDescription,
-    componentDescriptionToc = [],
-    classDescriptions,
-    propDescriptions,
-    slotDescriptions,
-  } = descriptions[userLanguage];
-  const description = t('api-docs.pageDescription').replace(/{{name}}/, componentName);
-
-  const source = filename
-    .replace(/\/packages\/(grid\/|)(.+?)?\/src/, (match, dash, pkg) => `@mui/${pkg}`)
-    // convert things like `/Table/Table.js` to ``
-    .replace(/\/([^/]+)\/\1\.(js|tsx)$/, '');
```

Comment:
@alexfauquette 
We miss this conversion: https://master--material-ui.netlify.app/x/api/data-grid/data-grid/#import

<img width="857" src="https://github.com/mui/mui-x/assets/13808724/3bdf6710-7c40-4fba-a3c7-86d839a243c8">

Reply:
Done in https://github.com/mui/material-ui/pull/38297